lua/ibl/init.lua

```diff
@@ -369,10 +389,67 @@ M.refresh = function(bufnr)
 
         last_whitespace_tbl = whitespace_tbl
 
-        -- #### make virtual text ####
-        local scope_start = row == scope_row_start
-        local scope_end = row == scope_row_end
-        if scope_start and scope then
+        --- current indent
+        if current_indent_enabled and row == lnum then
+            local current_indent_whitespace_tbl = vim.tbl_extend("keep", {}, whitespace_tbl)
+
+            while #current_indent_whitespace_tbl > 0 do
+                if indent.is_indent(current_indent_whitespace_tbl[#current_indent_whitespace_tbl]) then
+                    table.remove(current_indent_whitespace_tbl, #current_indent_whitespace_tbl)
+                    break
+                end
+                table.remove(current_indent_whitespace_tbl, #current_indent_whitespace_tbl)
+            end
+
+            current_indent_col_start_single = #current_indent_whitespace_tbl
+
+            for j = indent_state and #indent_state.stack or 0, 1, -1 do
```

Comment:
Is there a reason this is not:
```suggestion
            for j = indent_state and #indent_state.stack or 1, 1, -1 do
```
I am a bit confused about what happens when `indent_state` is nil.